dealroom_firestore_connector/__init__.py

```diff
@@ -396,24 +434,31 @@ def set_history_doc_refs(
     elif "final_url" in history_refs and len(history_refs["final_url"]) > 0:
         count_history_refs = len(history_refs["final_url"])
         key_found = "final_url"
-    elif "current_related_urls" in history_refs and len(history_refs["current_related_urls"]) > 0:
+    elif (
+        "current_related_urls" in history_refs
+        and len(history_refs["current_related_urls"]) > 0
+    ):
         count_history_refs = len(history_refs["current_related_urls"])
         key_found = "current_related_urls"
     else:
         count_history_refs = 0
+
     # CREATE: If there are not available documents in history
     if count_history_refs == 0:
         # Add any default values to the payload
         _payload = {
-            "dealroom_id": _NOT_IN_DEALROOM_ENTITY_ID,
-            "final_url": "",
+            "dealroom_id": finalurl_or_dealroomid
+            if is_dealroom_id
+            else _NOT_IN_DEALROOM_ENTITY_ID,
+            "final_url": finalurl_or_dealroomid if not is_dealroom_id else "",
```

Comment:
This covers the case where we we are creating orphans documents, with `final_url = ""` & `dealroom_id=-1`. Altough this happens in very edge cases we should handle it. A case such this is when we don't have a dealroom entity in Firestore for some reason and there is an update on that entity. This would create an orphan

Reply:
Leave a comment to explain that we handle the empty payload case